tests/backup/backup_helper.go

```diff
@@ -1254,6 +1292,213 @@ func IsBackupLocationPresent(bkpLocation string, ctx context.Context, orgID stri
 	return false, nil
 }
 
+// GetCurrentScheduleBackupCount returns the current number of schedule backups for the given schedule
+func GetCurrentScheduleBackupCount(ctx context.Context, scheduleName string, orgID string) (int, error) {
+	allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(ctx, scheduleName, orgID)
+	if err != nil {
+		return -1, err
+	}
+	return len(allScheduleBackupNames), nil
+}
+
+// GetOrdinalScheduleBackupName returns the name of the schedule backup at the specified ordinal position for the given schedule
+func GetOrdinalScheduleBackupName(ctx context.Context, scheduleName string, ordinal int, orgID string) (string, error) {
+	allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	if len(allScheduleBackupNames) == 0 {
+		return "", fmt.Errorf("no backups were found for the schedule [%s]", scheduleName)
+	}
+	if ordinal < 1 {
+		return "", fmt.Errorf("the ordinal value [%d] for schedule backups with schedule name [%s] is invalid. valid values range from 1 to [%d]", ordinal, scheduleName, len(allScheduleBackupNames))
+	}
+	if ordinal > len(allScheduleBackupNames) {
+		return "", fmt.Errorf("schedule backups with schedule name [%s] have not been created up to the provided ordinal value [%d]", scheduleName, ordinal)
+	}
+	return allScheduleBackupNames[ordinal-1], nil
+}
+
+// GetOrdinalScheduleBackupNameAfterCheck returns the name of the schedule backup at the specified ordinal position for the given schedule after success check
+func GetOrdinalScheduleBackupNameAfterCheck(ctx context.Context, scheduleName string, ordinal int, orgID string) (string, error) {
+	ordinalScheduleBackupName, err := task.DoRetryWithTimeout(func() (interface{}, bool, error) {
+		ordinalScheduleBackupName, err := GetOrdinalScheduleBackupName(ctx, scheduleName, ordinal, orgID)
+		if err != nil {
+			if ordinal < 1 {
+				return "", false, err
+			}
+			return "", true, err
+		}
+		return ordinalScheduleBackupName, false, nil
+	}, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second)
+	if err != nil {
+		return "", err
+	}
+	_, err = backupSuccessCheck(ordinalScheduleBackupName.(string), orgID, 0, 0, ctx)
+	if err != nil {
+		return "", err
+	}
+	return ordinalScheduleBackupName.(string), nil
+}
```

Comment:
There is no need for this function. We needs function which will return the backup name and UID. Once the test case gets the backup name, it is up to the test case what to do with it. To check the status of the backup, we simply have to inspect the backup name which is straightforward.

Reply:
This helps a bit in reducing code duplication. For instance, GetOrdinalScheduleBackupName can be used to retrieve the backup name of a previously created schedule backup by passing the ordinal. However, if we need to get the ordinal name of a future schedule backup only after it is successful, we can use GetOrdinalScheduleBackupNameAfterCheck.

Reply:
As discussed, we can remove this

Reply:
Removed GetOrdinalScheduleBackupNameAfterCheck.